internal/libyaml/emitter.go

```diff
@@ -902,77 +902,76 @@ func (emitter *Emitter) emitNode(event *Event,
 	case MAPPING_START_EVENT:
 		return emitter.emitMappingStart(event)
 	default:
-		return emitter.setEmitterError(
-			fmt.Sprintf("expected SCALAR, SEQUENCE-START, MAPPING-START, or ALIAS, but got %v", event.Type))
+		return fmt.Errorf("expected SCALAR, SEQUENCE-START, MAPPING-START, or ALIAS, but got %v", event.Type)
```

Comment:
nit: `EventType` has `func String()` function, so `%s` can be used in an error message instead `#v`, which uses reflect to get a type.
```suggestion
		return fmt.Errorf("expected SCALAR, SEQUENCE-START, MAPPING-START, or ALIAS, but got %s", event.Type)
```